Add options to configure SSH ciphers and algorithms for the eos_cli_config_gen role

dominikrappaport opened the pull request:
## Change Summary

Added options to configure ciphers and algorithms for SSH

Add option to configure
  - ciphers
  - key-exchange methods
  - MAC algorithms
  - Hostkey server algorithms

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)


## Component(s) name

`arista.avd.eos_cli_config_gen role`

## Proposed changes

Added new parameters for `management ssh` so users can configure ciphers and other algorithms.

## How to test
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->

Tested options (partial and full) and generated configlets.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

dominikrappaport committed:
Add options to configure SSH ciphers and algorithms for the eos_cli_config_gen role

Add option to configure
  - ciphers
  - key-exchange methods
  - MAC algorithms
  - Hostkey server algorithms

onurgashi commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:
```diff
@@ -15,6 +15,20 @@ management ssh
+   mac {{ management_ssh.mac|join(" ") }}
+{%     endif %}
+{%     if management_ssh.hostkey is arista.avd.defined %}
+{%     if management_ssh.hostkey.server is arista.avd.defined %}
```
This needs 4 spaces indentation, as it's a nested if.

dominikrappaport replied:
Sure thing!

onurgashi commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:
```diff
@@ -15,6 +15,20 @@ management ssh
+{%     if management_ssh.hostkey is arista.avd.defined %}
+{%     if management_ssh.hostkey.server is arista.avd.defined %}
+   hostkey server {{ management_ssh.hostkey.server|join(" ") }}
+{%     endif %}
```
This needs 4 spaces indentation, as it's a nested if.

onurgashi commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:
```diff
@@ -1696,6 +1696,19 @@ management_ssh:
       vrf: < vrf name >
   idle_timeout: < 0-86400 in minutes >
+  cipher:
+    - cipher1
```
< cipher1 > instead of cipher1 and so on

onurgashi commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:
```diff
@@ -1696,6 +1696,19 @@ management_ssh:
+    - cipher1
+    - cipher2
+  key-exchange:
+    - method1
```
< method1 > and so on

onurgashi commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:
```diff
@@ -1696,6 +1696,19 @@ management_ssh:
+    - method1
+    - method2
+  mac:
+    - mac_algorithm1
```
< mac_algorithm1 > and so on :D

onurgashi commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:
```diff
@@ -1696,6 +1696,19 @@ management_ssh:
+    - mac_algorithm2
+  hostkey:
+    server:
+      - algorithm1
```
< algorithm1 >

onurgashi commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:
```diff
@@ -15,6 +15,20 @@ management ssh
    idle-timeout {{ management_ssh.idle_timeout }}
 {%     endif %}
+{%     if management_ssh.cipher is arista.avd.defined %}
+   cipher {{ management_ssh.cipher|join(" ") }}
```
cipher | join(" "), spacing for all filters please.

onurgashi reviewed (changes requested):
I added some comments, if you could fix them it would be awesome! 👍

dominikrappaport committed:
Minor fixes as per request from Onur

- Correct indentation in Jinja2 templates
- Add space between `variable | filter` in Jinja2 templates
- Surround sample values in README.md with < >

onurgashi reviewed (approved):
Thank you @dominikrappaport LGTM!

dominikrappaport committed:
Update CI artifacts

titom73 reviewed (approved):
LGTM!
